src/check_pack.c

```diff
@@ -70,12 +72,13 @@ static size_t get_max_msg_size(void)
     return value;
 }
 
-/* typedef an unsigned int that has at least 4 bytes */
+/* typedef an unsigned int that has exactly 4 bytes, as required by pack_int() */
 typedef uint32_t ck_uint32;
+#define CK_UINT32_MAX UINT32_MAX
 
 
-static void pack_int(char **buf, int val);
-static int upack_int(char **buf);
+static void pack_int(char **buf, ck_uint32 val);
```

Comment:
I'm wondering what the value of having the ck_uint32 typedef is, rather than just using uint32_t directly. I'm not sure it buys anything. Same for UINT32_MAX vs CK_UINT32_MAX. What do you think?

Reply:
Well, `ck_uint32` was there when I came. :grin: 
I added `CK_UINT32_MAX` for the sake of consistence.
`ck_uint32` is only used in `check_pack.c` and only internally, not in the header. It would not be a big thing to refactor it away.